pkg/http/client.go

```diff
@@ -0,0 +1,62 @@
+package http
+
+import (
+	"crypto/tls"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/dcos/dcos-cli/pkg/config"
+)
+
+type Client struct {
+	baseURL    string
+	headers    http.Header
+	httpClient *http.Client
+}
+
+func New(conf config.Config) *Client {
+	var tlsConfig *tls.Config
+	if (config.TLS{}) == conf.TLS() {
+		tlsConfig = &tls.Config{InsecureSkipVerify: true}
+	} else {
+		tlsConfig = &tls.Config{InsecureSkipVerify: conf.TLS().Insecure, RootCAs: conf.TLS().RootCAs}
+	}
+	transport := &http.Transport{
+		TLSClientConfig: tlsConfig,
+	}
+	httpClient := &http.Client{
+		Transport: transport,
+		Timeout:   time.Duration(conf.Timeout()) * time.Second,
+	}
+
+	return &Client{conf.URL(), make(http.Header), httpClient}
+}
+
+func (c *Client) Get(endpoint string) (resp *http.Response, err error) {
+	req, err := http.NewRequest("GET", c.baseURL+endpoint, nil)
+	if err != nil {
+		return nil, err
+	}
+	return c.do(req)
+}
+
+func (c *Client) Post(endpoint string, body io.Reader) (resp *http.Response, err error) {
+	req, err := http.NewRequest("POST", c.baseURL+endpoint, body)
+	if err != nil {
+		return nil, err
+	}
+	return c.do(req)
+}
+
+func (c *Client) do(req *http.Request) (resp *http.Response, err error) {
+	req.Close = true
```

Comment:
What is the reason for setting this to false? This disallows re-using the same underlying TCP connection for future requests, which shouldn't be the case imo.